src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstructor.cs

```diff
@@ -0,0 +1,155 @@
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull => ImmutableHashSet<string>.Empty;
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations
+            => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
+
+        public override Symbol? AssociatedSymbol => null;
+
+        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+
+        internal override bool GenerateDebugInfo => false;
+
+        public override Symbol ContainingSymbol => ContainingType;
+
+        public override ImmutableArray<Location> Locations => ContainingType.Locations;
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Public;
+
+        public override bool IsStatic => false;
+
+        public override bool IsVirtual => false;
+
+        public override bool IsOverride => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => false;
+
+        public override bool IsExtern => false;
+
+        internal override LexicalSortKey GetLexicalSortKey()
+        {
+            // We need a separate sort key because struct records will have two synthesized
+            // constructors: the record constructor, and the parameterless constructor
+            return LexicalSortKey.SynthesizedRecordDeconstructor;
+        }
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+
+            Debug.Assert(Parameters.Length == _members.Length);
+            var statementsBuilder = ArrayBuilder<BoundStatement>.GetInstance(_members.Length + 1);
+            for (int i = 0; i < _members.Length; i++)
+            {
+                var parameter = Parameters[i];
+                var member = _members[i];
+                var propertyAccess = member switch
+                {
+                    PropertySymbol prop => F.Property(F.This(), prop),
+                    FieldSymbol field => F.Field(F.This(), field),
+                    _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                };
+
+                // parameter_i = property_i;
+                statementsBuilder.Add(F.Assignment(F.Parameter(parameter), propertyAccess));
```

Comment:
>Assignment [](start = 40, length = 10)

This doesn't apply any conversions that may be necessary from the type of the expression to the type of the parameter.

Also, we are missing an error in semantic analysis when such a conversion does not exist.

Reply:
I think this is avoidable by giving a binding error when the property has a different type than the parameter.

Reply:
Do we need a conversion even in cases where the parameter type matches the property type? For instance when overriding a property on a generic base with a constructed derived type.
```C#
record A<T>(T P);
record B(int P) : A<int>;
```

Reply:
I do not think so, but I could be missing something.